azurerm/internal/services/synapse/tests/synapse_role_assignment_resource_test.go

```
@@ -61,7 +61,10 @@ func testCheckAzureRMSynapseRoleAssignmentExists(resourceName string) resource.T
 		if err != nil {
 			return err
 		}
-		client := synapseClient.AccessControlClient(id.Workspace.Name, environment.SynapseEndpointSuffix)
+		client, err := synapseClient.AccessControlClient(id.Workspace.Name, environment.SynapseEndpointSuffix)
+		if err != nil {
+			return nil
```

whilst this may seem odd, this should return the error here